app/models/document.py

```diff
@@ -94,10 +100,18 @@ def status(self):
 
     @property
     def date_created(self):
```

Comment:
Should the `date_created` and `date_published` properties be columns in the table instead? Or is relying on the events table fine?